### PR TITLE
Feature/plcn 417 frontend user auth setup via makevar

### PR DESCRIPTION
**Pull request by itm-mne**
Setup of frontend auth flow. A registered user can now login and logout (user credentials are written in Pelican project Slack).

When a visitor is not logged in, only the login-page is accessible and writing any other URL redirects to the login page.

When a user is logged in, Clients, Contacts and suppliers are accessible either by clicking nav bar or navigation in the URL. Navigating to `/create-user` (the admin page) returns the 401 - Unauthorized page. If the user is an admin the admin page is accesible from both the navbar and URL.

_There is a known bug with the location filter, that are not a part of this PR._

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,73 @@
...
+									<PageContainer>
+										<Typography variant="h5">
+											Here you should hopefully be able to create a new user :)
+										</Typography>
+									</PageContainer>
+									// TODO: Transform "create-user" page into react component
+								}
+							/>
+						)}
+						<Route path="*" element={<NotFound404 />} />
```
Comment (AntonEsbjergIT-Minds) [inline src/App.tsx:47]: Should this not be an 401 Unauthorized instead? Or maybe simply reroute elsewhere

File: src/App.tsx [hunk 2]
```diff
@@ -1,13 +1,73 @@
...
+				<Routes>
+					<Route element={<WithNav />}>
+						<Route path="/" element={<WallOfClients />} />
+						<Route path="/contacts" element={<Contacts />} />
+						<Route path="/suppliers" element={<Suppliers />} />
+						<Route path="/recommendations" element={<Recommendations />} />
+						{currentUser.role === 'admin' && (
+							<Route
+								path="/create-user"
+								element={
```
Comment (AntonEsbjergIT-Minds) [inline src/App.tsx:37]: Seems like there is some glitch with this path here. It becomes inaccessible if I logout on the admin, login on a standard user, log out of that user and then log back in on the admin.

File: src/contexts/UserStore.tsx [hunk 3]
```diff
@@ -1,29 +1,64 @@
...
-const userVar = makeVar<ActiveUser>(activeUser);
+const userVar = makeVar<ActiveUser>(guestUser);
 
 const useLoginState = () => {
 	return useReactiveVar(userVar);
 };
 
 const setActiveUser = (user: ActiveUser) => {
+	switch (user.role) {
+		case 1:
```
Comment (AntonEsbjergIT-Minds) [inline src/contexts/UserStore.tsx:38]: Are these 1 and '1' cases necessary?

Comment (itm-mne) [reply src/contexts/UserStore.tsx:44]: I have seen role received as both "user", "1", and 1, so yes for now. Might be more reliable in the future and will be changed at that point.

Comment (Nicolai-Bak) [reply src/contexts/UserStore.tsx:44]: Would by a bit more readable this way maybe: 
``` ts
		case 1:
		case '1':
			user.role = 'user';
			break;
		case 2:
		case '2':
			user.role = 'admin';
			break;
```

**Review by Nicolai-Bak — APPROVED** [review]

**Pull request merged**